### ISSUE TITLE
[BUG] the version 1.2 introduced a regression

### ISSUE THREAD

**Issue by buenjybar** [react-dropzone/file-selector#106]
**Describe the bug**

When running the new version of `react-dropzone: @14.3.5` [link](https://github.com/react-dropzone/react-dropzone/releases/tag/v14.3.5), I noticed a regression introduced in this library

🐛 This bug was introduced in this [commit](https://github.com/react-dropzone/file-selector/commit/11d34a75f1dfa668e3da8c62911ce7aead224619#diff-b1b6505efe0d05e629847160e4e4c7ca59d9828ea5023fad78632e0a325debd7R127)

The actual error is `FileDrop-ky6DIxvn.js:5`
```
TypeError: Cannot read properties of undefined (reading 'getFile')
    at FileDrop-ky6DIxvn.js:1:43484
    at Generator.next (<anonymous>)
    at FileDrop-ky6DIxvn.js:1:386
    at new Promise (<anonymous>)
    at E (FileDrop-ky6DIxvn.js:1:183)
    at FileDrop-ky6DIxvn.js:1:43435
    at async Promise.all (__/index 0)
```

Accordingly to the experimental documentation of **DataTransferItem: getAsFileSystemHandle() method** [MDN](https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/getAsFileSystemHandle)
> A [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
If the item's [kind](https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/kind) property is "file", and this item is accessed in the [dragstart](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dragstart_event) or [drop](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/drop_event) event handlers, then the returned promise is fulfilled with a [FileSystemFileHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileHandle) if the dragged item is a file or a [FileSystemDirectoryHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle) if the dragged item is a directory.
**Otherwise, the promise fulfills with null.**

The return value can be a promise or `null`, but the current code doesn't manage `null` values

**Expected behavior**
This shouldn't crash at runtime 

**Desktop (please complete the following information):**
- userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36

**Additional context**

**It seems that the old/legacy behavior is working as expected.**

**Comment by jonkoops** [comment]
@rolandjitsu should we set up a `v1` branch so we can fix this one without creating a conflict with the `beta` branch?

**Comment by rolandjitsu** [comment]
That's a bug alright. Thanks @buenjybar for reporting it. @jonkoops yes, we'll need a `1.2.x` branch where we cherry pick a commit from `main` after we make a PR to `main` with the same fix for v2.

**Comment by jonkoops** [comment]
@buenjybar can you provide some details on how this can be reproduced? Which browser is being used, and in what way is the library called? Please provide a minimal code example like we have in the README.

**Comment by buenjybar** [comment]
Sure the scenario is being done in a cypress e2e tests, I will try to find the best way to reproduce this bug

**Comment by jonkoops** [comment]
Great thanks! Please make sure to provide a minimal and reproducible example that does not include any third party libraries/frameworks such as Cypress, React, etc.

**Comment by buenjybar** [comment]
you can see how to reproduce the bug here.
👉 https://github.com/buenjybar/react-dropzone-issue106/

**Comment by jonkoops** [comment]
That example still contains a bunch of framework specific code, could you provide a minimal example without React and testing frameworks and only `file-selector`? Here is [a StackBlitz project](https://stackblitz.com/edit/vitejs-vite-7gz427?file=src%2Fmain.ts) you can use as a base.

**Comment by buenjybar** [comment]
> That example still contains a bunch of framework specific code, could you provide a minimal example without React and testing frameworks and only `file-selector`? Here is [a StackBlitz project](https://stackblitz.com/edit/vitejs-vite-7gz427?file=src%2Fmain.ts) you can use as a base.

In the code sample there is nothing sophisticated. 
I have explained how to reproduce it in the [README.md](https://github.com/buenjybar/react-dropzone-issue106/blob/master/my-app/README.md)


```
$ npm install
$ npm run start
// in a different terminal
$ npm run cy:open

// on cypress
Click on "E2E Testing"
Run on Chrome v.130 (whatever)
Run the app.spec.cy.js
open the console and see the crash
```

```
TypeError: Cannot read properties of undefined (reading 'getFile')
    at FileDrop-ky6DIxvn.js:1:43484
    at Generator.next (<anonymous>)
    at FileDrop-ky6DIxvn.js:1:386
    at new Promise (<anonymous>)
    at E (FileDrop-ky6DIxvn.js:1:183)
    at FileDrop-ky6DIxvn.js:1:43435
    at async Promise.all (__/index 0)
```

**Comment by buenjybar** [comment]
I have added this to reproduce the issue https://github.com/react-dropzone/file-selector/pull/108

**Comment by jonkoops** [comment]
@buenjybar I did a bit of refactoring, could you tell me if #114 fixes the problem for you?

**Comment by rolandjitsu** [comment]
I've also filed https://github.com/cypress-io/cypress/issues/30629 to see if the Cypress team can provide any insight.

**Comment by rolandjitsu** [comment]
Hmmm ... it seems like this may be a browser issue. If you try the following in Chrome 130:
```js
const dt = new document.defaultView.DataTransfer();
const js = JSON.stringify({test: 1});
const b = new Blob([js], {type: "application/json"});
const f = new File([b], "test.json");
dt.items[0].getAsFileSystemHandle().then(h => console.log(h))
```

You'll get `undefined` logged.

I'm not sure what that means, but that's where the problem is. We expect a file or dir handle, but we get undefined. 

I mean, it is an experimental API, so it's to be expected. But in Chrome, the API should be available and work according to the compatibility table.

**Comment by buenjybar** [comment]
I can confirm the Fix in 2.1.2 👍  
Thank you @rolandjitsu and @jonkoops
